### ISSUE TITLE
All ESMF tutorial example

### ISSUE THREAD

**Issue by bena-nasa** [GEOS-ESM/MAPL#1848]
After going through the MAPL tutorials with a new user, myself as well as apparently @aoloso and @mathomp4 all independently came to the same idea that perhaps an all ESMF version (i.e. only pure Fortran or any API from a library in baselines, no MAPL anything) of one of them, (maybe the single component that has a couple fields), would be a useful learning tool. We could point to the MAPL version, then point to the corresponding ESMF only and say here is what you would have to do with only ESMF and this is what MAPL has abstracted out is hidden from the user. This way the user could explicit see here is what we are hiding and would probably help.

Even for a simple case like one grid comp, and not having access  to our cap or grid creation machinery, it on non-trivial amount of code (not a crazy amount, but I would think it would take me a concentrated day to code it from scratch). I think this should be enough to get the point across as a start.

**Comment by tclune** [comment]
I agree.  The trick will be to identify the right chunk, or more likely a collection of chunks.    We don't want to seek out the worst cases - that would not be fair.   We want realistic examples to remind other GEOS developers why we have jobs.  :-)

**Comment by metdyn** [comment]
@bena-nasa: Your proposal:  a tutorial case  without v.s. with MAPL software interface when calling ESMF will be illustrative and very helpful to users. 

A related question to @mathomp4: 
- if someone from another HPC center has a loaded library which can build and run ESMF, what is the minimal additional library set needed to build and run MAPL?

**Comment by mathomp4** [comment]
> @bena-nasa: Your proposal: a tutorial case without v.s. with MAPL software interface when calling ESMF will be illustrative and very helpful to users.
> 
> A related question to @mathomp4:
> 
> * if someone from another HPC center has a loaded library which can build and run ESMF, what is the minimal additional library set needed to build and run MAPL?

@metdyn Well, I'd be most confident with what is in here https://github.com/GEOS-ESM/MAPL/releases/tag/v2.32.0. You can see the libraries in Baselibs 7.7.0

If you have ESMF you probably have netCDF, etc. It's mainly the GFE libraries you'd need (though not all are required depending on how you want to build MAPL)

**Comment by metdyn** [comment]
@mathomp4:  Great, thank you for the information! I understand the GFE now. I will give a try to build. 

You are right, when I load the esmf library I see the following minimal module on NCAR computers:
  1) ncarenv/1.3   3) ncarcompilers/0.5.0   5) openmpi/4.1.1
  2) gnu/10.1.0    4) netcdf/4.8.1          6) esmf_libs/8.2.0

**Comment by bena-nasa** [comment]
Started a branch where I'll do this here and there when I need a break, going to make a single cap-like driver main that will time look and a single component with a few exports run by said driver. That should get the point across for comparison

**Comment by bena-nasa** [comment]
@mathomp4 @tclune @atrayano 

After our discussion today sounds like the tutorial code that exists now is insufficient. I'll take this up again and add more. Sounds like we all agree a pure ESMF tutorial would be useful. I've started a branch for this here I will work on:
https://github.com/GEOS-ESM/MAPL/tree/feature/bmauer/pure_esmf_tut

Please let me know what other MAPL concepts I should create tutorials for.
